zerver/tests/test_email_notifications.py

```diff
@@ -941,6 +942,40 @@ def test_relative_to_full_url(self) -> None:
                           'https://www.google.com/images/srpr/logo4w.png</a></p>'
         self.assertEqual(actual_output, expected_output)
 
+    def test_spoilers_in_html_emails(self) -> None:
+        # Test HTML spoilers that use fix_spoilers()
+        test_data = "<div class=\"spoiler-block\"><div class=\"spoiler-header\">\n\n<p><a>header</a> text</p>\n</div><div class=\"spoiler-content\" aria-hidden=\"true\">\n\n<p>content</p>\n</div></div>\n\n<p>outside spoiler</p>"
+        actual_output = fix_spoilers_in_html(test_data, 'en')
+        expected_output = "<div><div class=\"spoiler-block\">\n\n<p><a>header</a> text<span> </span><span class=\"spoiler-title\" title=\"Open Zulip to see the spoiler content\">(Open Zulip to see the spoiler content)</span></p>\n</div>\n\n<p>outside spoiler</p></div>"
+        self.assertEqual(actual_output, expected_output)
+
+        # test against our markdown_test_cases so these features do not get out of sync.
+        fixtures = ujson.loads(self.fixture_data("markdown_test_cases.json"))
+        test_fixtures = {}
+        for test in fixtures['regular_tests']:
+            if 'spoiler' in test['name']:
```

Comment:
I'm not sure how much value these tests add but its a net positive. Still, we should have a better way of marking which tests to run; this is prone to renames in markdown_test_cases passing silently.